app/controllers/recipes_controller.rb

```diff
@@ -21,6 +21,36 @@ def create
     end
   end
 
+  def edit
+    if current_user.id == Recipe.find(params[:id]).user.id
```

Comment:
It's possible here to reduce the number of db queries looking up on the recipes table without sacrificing security. Currently line 25 and line 26 both lookup on the recipes table almost identically.